app/code/Magento/Indexer/Model/ProcessManager.php

```diff
@@ -135,11 +148,20 @@ private function isSetupMode(): bool
      */
     private function startChildProcess(callable $userFunction)
     {
-        // phpcs:ignore Magento2.Functions.DiscouragedFunction
-        $status = call_user_func($userFunction);
-        $status = is_int($status) ? $status : 0;
-        // phpcs:ignore Magento2.Security.LanguageConstruct.ExitUsage
-        exit($status);
+        try {
+            // phpcs:ignore Magento2.Functions.DiscouragedFunction
+            $status = call_user_func($userFunction);
```

Comment:
Can we remove call_user_func here? Like this
```suggestion
            $status = $userFunction();
```

Reply:
Wouldn't be really backwards compatible as call_user_func can also have strings passed in the form of \Foo::test, which would throw an exception in the suggested case.